public/app/features/alerting/unified/utils/rule-id.ts

```diff
@@ -104,6 +104,28 @@ export function equal(a: RuleIdentifier, b: RuleIdentifier) {
     );
   }
 
+  // It might happen to compare Cloud and Prometheus identifiers for datasources with available Ruler API
+  // It happends when the Ruler API timeouts and the UI cannot create Cloud identifiers, so it creates a Prometheus identifier instead.
+  if (isCloudRuleIdentifier(a) && isPrometheusRuleIdentifier(b)) {
```

Comment:
(from our earlier chat - would be good to refactor this logic in a separate PR so that we check if it's cloud, or prom, and then we have a list of ordered properties that we check. That way the different `if` blocks collapse down to:
`check the relevant properties from "a" against the relevant properties from "b"`, and makes it easier to see what properties should be checked with what)